File: week01/day07/backend/routers/session.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session as DbSession

from schemas.session import SessionCreate, MessageAdd, ConfirmSubmit
from dependencies import get_db, get_current_worker
from services import session as service
from utils.response import ok

router = APIRouter(prefix="/api", tags=["AI 对话"])


@router.get("/worker/patients")
def get_worker_patients(
    db: DbSession = Depends(get_db),
    worker=Depends(get_current_worker),
):
    result = service.get_worker_patients(db, worker.id)
    return ok(result)


@router.post("/sessions")
def create_session(
    data: SessionCreate,
    db: DbSession = Depends(get_db),
    worker=Depends(get_current_worker),
):
    result = service.get_or_create_session(db, data, worker.id)
    return ok(result, "对话已创建")


@router.get("/sessions/{session_id}")
def get_session(
    session_id: int,
    db: DbSession = Depends(get_db),
    worker=Depends(get_current_worker),
):
    result = service.get_session(db, session_id, worker.id)
    return ok(result)


@router.post("/sessions/{session_id}/messages")
def add_message(
    session_id: int,
    data: MessageAdd,
    db: DbSession = Depends(get_db),
    worker=Depends(get_current_worker),
):
    result = service.add_message(db, session_id, data, worker.id)
    return ok(result)


@router.post("/sessions/{session_id}/extract")
def extract_info(
    session_id: int,
    db: DbSession = Depends(get_db),
    worker=Depends(get_current_worker),
):
    result = service.extract_info(db, session_id, worker.id)
    return ok(result)


@router.post("/sessions/{session_id}/confirm")
def confirm_submit(
    session_id: int,
    data: ConfirmSubmit,
    db: DbSession = Depends(get_db),
    worker=Depends(get_current_worker),
):
    result = service.confirm_submit(db, session_id, data, worker.id)
    return ok(result, "信息已提交")

File: week01/day07/frontend/src/services/session.ts
import { request } from './request';
import type { ApiResponse } from '../types/api';
import type { Session, WorkerPatient, SessionResponse, AddMessageResponse, ExtractResult } from '../types/session';

export function getWorkerPatients(): Promise<ApiResponse<WorkerPatient[]>> {
  return request<ApiResponse<WorkerPatient[]>>({
    method: 'GET',
    url: '/api/worker/patients',
  });
}

export function createSession(patient_id: number): Promise<ApiResponse<Session>> {
  return request<ApiResponse<Session>>({
    method: 'POST',
    url: '/api/sessions',
    body: { patient_id },
  });
}

export function getSession(sessionId: number): Promise<ApiResponse<SessionResponse>> {
  return request<ApiResponse<SessionResponse>>({
    method: 'GET',
    url: `/api/sessions/${sessionId}`,
  });
}

export function addMessage(sessionId: number, content: string): Promise<ApiResponse<AddMessageResponse>> {
  return request<ApiResponse<AddMessageResponse>>({
    method: 'POST',
    url: `/api/sessions/${sessionId}/messages`,
    body: { content },
  });
}

export function extractInfo(sessionId: number): Promise<ApiResponse<ExtractResult>> {
  return request<ApiResponse<ExtractResult>>({
    method: 'POST',
    url: `/api/sessions/${sessionId}/extract`,
  });
}

export function confirmSubmit(sessionId: number, data: ExtractResult): Promise<ApiResponse<unknown>> {
  return request<ApiResponse<unknown>>({
    method: 'POST',
    url: `/api/sessions/${sessionId}/confirm`,
    body: data,
  });
}
